core/src/SYCL/Kokkos_SYCL_Team.hpp

```diff
@@ -175,6 +182,53 @@ class SYCLTeamMember {
     m_item.barrier(sycl::access::fence_space::local_space);
   }
 
+  // FIXME_SYCL move somewhere else and combine with other places that do
+  // parallel_scan
+  // Exclusive scan returning the total sum, compare
+  // https://developer.nvidia.com/gpugems/gpugems3/part-vi-gpu-computing/chapter-39-parallel-prefix-sum-scan-cuda
```

Comment:
We need to figure out what the license of this is.

Reply:
Oh, I can easily rewrite this to look more similar to what we have elsewhere if that's a concern.

Reply:
I rewrote it to look more similar to `RangePolicy::parallel_scan".